src/app/modules/spcp/spcp.oidc.client.ts

```diff
@@ -8,77 +9,90 @@ import {
   SignJWT,
 } from 'jose'
 import jwkToPem from 'jwk-to-pem'
-import { BaseClient } from 'openid-client'
+import { BaseClient, TokenSet } from 'openid-client'
 import { ulid } from 'ulid'
+import { URLSearchParams } from 'url'
 
-import { SpOidcClientCache } from './sp.oidc.client.cache'
+import { FormAuthType } from '../../../../shared/types'
+
+import { SpcpOidcBaseCilentCache } from './spcp.oidc.client.cache'
 import {
   CreateAuthorisationUrlError,
   CreateJwtError,
   ExchangeAuthTokenError,
   GetDecryptionKeyError,
+  GetSigningKeyError,
   GetVerificationKeyError,
   InvalidIdTokenError,
   JwkError,
   MissingIdTokenError,
   VerificationKeyError,
-} from './sp.oidc.client.errors'
+} from './spcp.oidc.client.errors'
 import {
+  CpOidcClientConstructorParams,
   CryptoKeys,
   SigningKey,
+  SpcpOidcBaseClientConstructorParams,
   SpOidcClientConstructorParams,
-} from './sp.oidc.client.types'
+} from './spcp.oidc.client.types'
 import {
   extractNricFromParsedSub,
+  isCPJWTVerifyResult,
   isEC,
   isECPrivate,
   isSigningKey,
   parseSub,
-} from './sp.oidc.util'
+} from './spcp.oidc.util'
 
 /**
- * Wrapper around the openid-client library to carry out authentication related tasks with Singpass NDI,
+ * Wrapper around the openid-client library to carry out authentication related tasks with Singpass and Corppass NDI,
  * and provides methods for decryption and verification of JWE/JWS returned by NDI after authorisation code exchange
+ * @parent for SpOidcClient and CpOidcClient classes
  */
-export class SpOidcClient {
-  #spOidcRpSecretKeys: CryptoKeys
-  #spOidcRpPublicKeys: CryptoKeys
-  #spOidcRpRedirectUrl: string
+export class SpcpOidcBaseClient {
+  #rpSecretKeys: CryptoKeys
+  #rpPublicKeys: CryptoKeys
+  #rpRedirectUrl: string
+  #rpClientId: string
+  #authType: FormAuthType.SP | FormAuthType.CP
 
   /**
    * @private
    * accessible only for testing
    */
-  _spOidcClientCache: SpOidcClientCache
+  _spcpOidcBaseCilentCache: SpcpOidcBaseCilentCache
```

Comment:
Typo in member as well (and in accesses below in the file)
```suggestion
  _spcpOidcBaseClientCache: SpcpOidcBaseClientCache
```

Reply:
Fixed in f7919072bf9508d1d36d8bd709977a4d2b036d2f